### PR TITLE
implement withdraw all

### PR DESCRIPTION
**Pull request by philipkcl**
# Withdrawing a journal with a continuation should prompt the user to withdraw the connected journal as well

*Please don't delete any sections when completing this PR template; instead enter **N/A** for checkboxes or sections which are not applicable, unless otherwise stated below*

See #
* https://github.com/DOAJ/doajPM/issues/3660

Describe the scope/purpose of the PR here in as much detail as you like

## Categorisation

This PR...
- [ ] has scripts to run
- [ ] has migrations to run
- [ ] adds new infrastructure
- [ ] changes the CI pipeline
- [ ] affects the public site
- [x] affects the editorial area
- [ ] affects the publisher area
- [ ] affects the monitoring

## Basic PR Checklist

Instructions for developers:
* For each checklist item, if it is N/A to your PR check the N/A box
* For each item that you have done and confirmed for yourself, check Developer box (including if you have checked the N/A box)

Instructions for reviewers:
* For each checklist item that has been confirmed by the Developer, check the Reviewer box if you agree
* For multiple reviewers, feel free to add your own checkbox with your github username next to it if that helps with review tracking

### Code Style

- No deprecated methods are used
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer

- No magic strings/numbers - all strings are in `constants` or `messages` files
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- ES queries are wrapped in a Query object rather than inlined in the code
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- Where possible our common library functions have been used (e.g. dates manipulated via `dates`)
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- Cleaned up commented out code, etc
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer

### Testing

- Unit tests have been added/modified
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer
  
- Functional tests have been added/modified
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer
  
- Code has been run manually in development, and functional tests followed locally
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer

- Have CSS/style changes been implemented?  If they are of a global scope (e.g. on base HTML elements) have the downstream impacts of the change in other areas of the system been considered?
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer

### Documentation

- FeatureMap annotations have been added
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer
  
- Documentation updates - if needed - have been identified and prepared for inclusion into main documentation (e.g. added and highlighted/commented as appropriate to this PR)
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer
  
- Core model documentation has been added to if needed: https://docs.google.com/spreadsheets/d/1lun2S9vwGbyfy3WjIjgXBm05D-3wWDZ4bp8xiIYfImM/edit
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer

- Events and consumers documentation has been added if needed: https://docs.google.com/spreadsheets/d/1oIeG5vg-blm2MZCE-7YhwulUlSz6TOUeY8jAftdP9JE/edit
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer
  
- The docs for this branch have been generated and pushed to the doc site (see docs/README.md for details)
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer

### Release Readiness

- If needed, migration has been created and tested locally
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer

- Release sheet has been created, and completed as far as is possible https://docs.google.com/spreadsheets/d/1Bqx23J1MwXzjrmAygbqlU3YHxN1Wf7zkkRv14eTVLZQ/edit#gid=964781844
  - [ ] N/A
  - [ ] Developer
  - [x] Reviewer

- There has been a recent merge up from `develop` (or other base branch).  List the dates of the merges up from develop below
  - 2023-08-18

## Testing

List the Functional Tests that must be run to confirm this feature

1. ...
2. ...

## Deployment

What deployment considerations are there? (delete any sections you don't need)

### Configuration changes

What configuration changes are included in this PR, and do we need to set specific values for production

### Scripts

What scripts need to be run from the PR (e.g. if this is a report generating feature), and when (once, regularly, etc).

### Migrations

What migrations need to be run to deploy this

### Monitoring

What additional monitoring is required of the application as a result of this feature

### New Infrastructure

What new infrastructure does this PR require (e.g. new services that need to run on the back-end).

### Continuous Integration

What CI changes are required for this

### CODE REVIEW

File: portality/bll/services/journal.py [hunk 1]
```diff
@@ -145,16 +151,20 @@ def csv(self, prune=True):
...
+            tmpStore.delete_file(container_id,
+                                 filename)  # don't delete the container, just in case someone else is writing to it
 
         action_register = []
         if prune:
             def sort(filelist):
                 rx = "journalcsv__doaj_(.+?)_utf8.csv"
-                return sorted(filelist, key=lambda x: datetime.strptime(re.match(rx, x).groups(1)[0], FMT_DATETIME_SHORT), reverse=True)
+                return sorted(filelist,
+                              key=lambda x: datetime.strptime(re.match(rx, x).groups(1)[0], FMT_DATETIME_SHORT),
```
Comment (RK206) [inline portality/bll/services/journal.py:162]: should dates library need to be used instead datetime

**Review by RK206 — APPROVED** [review]
Approved with minor comment

**Comment by RK206** [comment]
@philipkcl can you please check if dates can be used instead of datetime? Also can you please resolve the conflict for maned_form.yml

File: portality/bll/services/journal.py (hunk 1 above)
Comment (philipkcl) [reply portality/bll/services/journal.py:162]: did you mean replace `datetime.strptime` to `dates.parse`? 
behavior of `datetime.strptime` and `dates.parse` is different, `dates.parse` will auto guess the format of date.
`datetime.strptime` is low level, faster and more predictable to parse the date, we can open for developer to choice which method is better in this case.

**Review by RK206 — APPROVED** [review]

**Review by Steven-Eardley — APPROVED** [review]

**Comment by dommitchell** [comment]
@RK206 are there functional tests for this release? Thanks!

**Comment by philipkcl** [comment]
PR updated for feedback https://github.com/DOAJ/doajPM/issues/3660#issuecomment-1767763064

**Comment by philipkcl** [comment]
updated PR for https://github.com/DOAJ/doajPM/issues/3660#issuecomment-1812576470

**Comment by RK206** [comment]
Code looks good and working fine.
@philipkcl though I am able to test the functionality, I think we need to have feature tests for others to test.
Can you please add feature tests and release sheet?

**Comment by philipkcl** [comment]
@RK206 , it had functional test before but the functional test removed since the requirement change.

In current requirement is just display list of journal on popup.

I will add release sheet later.

image

**Comment by philipkcl** [comment]
@RK206 , release sheet already exist

**Review by RK206 — APPROVED** [review]

**Review by Steven-Eardley — APPROVED** [review]
A bit more macro magic than usual, but I can get behind it

**Pull request merged**